zyte_spider_templates/utils.py

```diff
@@ -15,8 +15,12 @@
 _URL_PATTERN = r"^https?://[^:/\s]+(:\d{1,5})?(/[^\s]*)*(#[^\s]*)?$"
 
 
-def get_domain(url: str) -> str:
-    return re.sub(r"^www\d*\.", "", parse_url(url).netloc)
+def get_domain(url: str, include_port: bool = True) -> str:
```

Comment:
I don't really like this but this seems like the easiest way to fix both middlewares, as `AllowOffsiteMiddleware` explicitly doesn't support ports while `OffsiteRequestsPerSeedMiddleware` uses `url_is_from_any_domain()` which  compares the ports.